src/app/components/hris/employees/employee-profile/shared-accordion-functionality.ts

```diff
@@ -225,15 +228,37 @@ export class SharedAccordionFunctionality {
     for (const controlName in formControls) {
       if (formControls.hasOwnProperty(controlName)) {
         const control = formControls[controlName];
-        if (control.value != null && control.value != '' && this.hasDisability != false && control.value != "na") {
-          numberOfPopulatedFields++;
-        }
-        else if (controlName.includes("disability") && this.hasDisability == false) {
-          numberOfPopulatedFields++;
+
+        if (control.validator) {
+          const validator = control.validator({} as AbstractControl);
+          isRequired = validator && validator['required'] ? true : false;
         }
+
+        if (isRequired) {
+          if(this.hasDisability !== false ){
+           numberOfRequiredFields=5;
+          }else{
+            numberOfRequiredFields=3;
+          }
```

Comment:
Refactor: the readability of this if statement could be improved perhaps with some whitespace before and after the brackets, and some new lines.

if (condition) {
   //some cool thing
}
else {
  //some other thing
}